lms/services/dashboard.py

```diff
@@ -246,13 +275,68 @@ def get_course_roster(
                 # For launch data we always add the "active" column as true for compatibility with the roster query.
             ).add_columns(true())
 
+        return roster_last_updated, query.order_by(LMSUser.display_name, LMSUser.id)
+
+    def get_segments_roster(
+        self, segments: list[LMSSegment], h_userids: list[str] | None = None
+    ) -> tuple[datetime | None, Select[tuple[LMSUser, bool]]]:
+        """Return a query that fetches the roster for a segment."""
+        assert (
```

Comment:
This is now enforced by the UI but just keeping this here to remember this assumption.

Reply:
It makes sense to enforce domain rules server-side, even if the UI would usually prevent invalid combinations.